src/hotspot/cpu/riscv/templateTable_riscv.cpp

```diff
@@ -280,7 +280,14 @@ void TemplateTable::bipush() {
 
 void TemplateTable::sipush() {
   transition(vtos, itos);
-  __ load_unsigned_short(x10, at_bcp(1));
+  if (AvoidUnalignedAccesses) {
+    __ lbu(t1, at_bcp(1));
+    __ lbu(x10, at_bcp(2));
+    __ slli(x10, x10, 8);
+    __ add(x10, t1, x10);
+  } else {
+    __ load_unsigned_short(x10, at_bcp(1));
+  }
   __ revb_w_w(x10, x10);
   __ sraiw(x10, x10, 16);
```

Comment:
I think we can further simplify this sequence into something like:
```
if (AvoidUnalignedAccesses) {
  __ load_signed_byte(x10, at_bcp(1));
  __ load_unsigned_byte(t1, at_bcp(2));
  __ slli(x10, x10, 8);
  __ add(x10, x10, t1);
} else {
  __ load_unsigned_short(x10, at_bcp(1));
  __ revb_w_w(x10, x10);
  __ sraiw(x10, x10, 16);
}
```

Reply:
why have you replaced revb_w_w with revb_h_h ?

Reply:
> why have you replaced revb_w_w with revb_h_h ?

Because it's cheaper in respect of number of instructions emitted when Zbb extension is not available and achieves the same functionality.

Reply:
A thanks, took time to understand the  __ revb_w_w(x10, x10); __ sraiw(x10, x10, 16); was kind of too much here